invenio_drafts_resources/services/records/components/files_base.py

```diff
@@ -0,0 +1,239 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-Drafts-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Records service component base classes."""
+from invenio_i18n import gettext as _
+from invenio_records_resources.services.files.transfer import TransferType
+from invenio_records_resources.services.records.components import (
+    FilesOptionsComponent,
+    BaseRecordFilesComponent,
+)
+from marshmallow import ValidationError
+
+from .base import ServiceComponent
+
+
+class RecordFilesComponent(ServiceComponent, BaseRecordFilesComponent):
+    def __init__(self, service, *args, **kwargs):
+        """Constructor."""
+        super().__init__(service)
+        self.files_component = FilesOptionsComponent(service)
+
+    #
+    # API
+    #
+    def create(self, identity, data=None, record=None, errors=None):
+        """Assigns files.enabled.
+
+        NOTE: `record` actually refers to the draft
+              (this interface is used in records-resources and rdm-records)
+        """
+        draft = record
+        files = self.get_record_files(draft)
+        enabled = data.get(self.files_data_key, {}).get("enabled", True)
+
+        if files.enabled != enabled:
+            if not self.service.check_permission(
+                identity, "manage_files", record=draft
```

Comment:
media files do not have the "manage_files" permission, if you can edit draft, you can enable/disable media_files too since they are additional, we don't protect them the same way as metadata-only